quickwit/quickwit-config/src/index_config/serialize.rs

```diff
@@ -61,6 +61,50 @@ pub fn load_index_config_from_user_config(
     index_config_for_serialization.build_and_validate(Some(default_index_root_uri))
 }
 
+/// Parses and validates an [`IndexConfig`] update. Ensures that the new
+/// configuration is valid in itself and compared to the current index config.
+/// If the new configuration omits some fields, the default values will be used,
+/// not those of the current index config. The only exception is the index_uri
+/// because it cannot be updated.
+pub fn load_index_config_update(
```

Comment:
The intent of this function would be clearer if the load was performed outside and it was just:
```rust
pub fn validate_index_config_update(
  current_index_config: &IndexConfig,
  new_index_config: &IndexConfig,
) -> anyhow::Result<()> {
}
```

Reply:
Yes, agreed, but for consistency with `load_index_config_from_user_config` I'll leave the parsing and validation together for now.

Reply:
I don't understand the consistency argument. Can you expand? What's wrong with:
```rust
let new_index_config = load_index_config_from_user_config(...)?;
let current_index_config = metastore.index_metadata(...)?.index_config;
validate_index_config_update(current_index_config, new_index_config)?;
metastore.update_index_config(..., new_index_config)?;
```